### PR TITLE
CLIReturnError expose return_code, stderr and msg to assertRaisesRegex

### LINKED ISSUE #3782: Fiz CLIError and CLIReturnError to enable assertRaiseRegex
Currently assertRaiseRegex can not be used because stderr is not present on CLIReturnError repr. Python 2 uses a `message` attribute to define exception msgs and Python 3 uses list attribute `args`.

### PR DESCRIPTION
**Pull request by renzon**
close #3782 

Test results

```console
============================= test session starts ==============================
platform linux2 -- Python 2.7.12, pytest-2.9.2, py-1.4.31, pluggy-0.3.1
rootdir: /home/renzo/PycharmProjects/robottelo, inifile: 
plugins: cov-2.3.0, xdist-1.14
collected 185 items

test_api_utils.py ..
test_cli.py ......................
test_config_casts.py ..............
test_config_settings.py .....
test_datafactory.py .....
test_decorators.py ...........................................
test_hammer.py .........
test_helpers.py ...........
test_host_info.py ................
test_ssh.py ..............
test_ui.py .....
test_ui_locators.py ........................
test_vm.py ......
decorators/test_host.py .........

========================== 185 passed in 1.72 seconds ==========================

Process finished with exit code 0
```

### CODE REVIEW

**Comment by renzon** [comment]
@pgagne @pondrejk after this got merged we can use self.assertRaisesRegexp to assert error msgs coming from CLI. Thanks @rochacbruno for the tip.

File: robottelo/cli/base.py [hunk 1]
```diff
@@ -29,7 +29,11 @@ def __init__(self, return_code, stderr, msg):
         self.message = msg
 
     def __str__(self):
-        return self.message
+        return repr(self)
```
Comment (rochacbruno) [inline robottelo/cli/base.py]: Please include some docstring here explaining the reason of this override.

File: robottelo/cli/base.py [hunk 2]
```diff
@@ -29,7 +29,11 @@ def __init__(self, return_code, stderr, msg):
...
+
+    def __repr__(self):
+        tmpl = u'CLIReturnCodeError(return_code={!r}, stderr={!r}, msg={!r}'
```
Comment (rochacbruno) [inline robottelo/cli/base.py]: same here

**Comment by rochacbruno** [comment]
ACK (commented a small docstring nitpick)

**Comment by oshtaier** [comment]
ACK

File: tests/robottelo/test_cli.py [hunk 3]
```diff
@@ -248,7 +258,17 @@ def test_init(self):
...
+        with self.assertRaisesRegexp(CLIReturnCodeError, u'1'):
+            raise CLIReturnCodeError(1, u'stderr', u'umsg')
+
+    def test_stderr_is_exposed(self):
+        """Check if stderr is exposed to assertRaisesRegexp"""
+        with self.assertRaisesRegexp(CLIReturnCodeError, u'stderr'):
+            raise CLIReturnCodeError(1, u'stderr', u'msg')
+
+    def test_message_is_exposed(self):
+        """Check if return_code is exposed to assertRaisesRegexp"""
```
Comment (sghai) [inline tests/robottelo/test_cli.py]: s/return_code/message ?

File: tests/robottelo/test_cli.py [hunk 4]
```diff
@@ -248,7 +258,17 @@ def test_init(self):
...
         self.assertEqual(error.message, error.msg)
 
-    def test_str(self):
-        """Check __str__ returns message attribute"""
-        error = CLIReturnCodeError(1, u'stderr', u'msg')
-        self.assertEqual(error.message, str(error))
+    def test_return_code_is_exposed(self):
+        """Check if return_code is exposed to assertRaisesRegexp"""
+        with self.assertRaisesRegexp(CLIReturnCodeError, u'1'):
+            raise CLIReturnCodeError(1, u'stderr', u'umsg')
```
Comment (oshtaier) [inline tests/robottelo/test_cli.py]: `(1, u'stderr', u'msg')`

**Comment by oshtaier** [comment]
ACK

**Pull request merged**